crates/apub_objects/src/utils/functions.rs

```diff
@@ -250,6 +253,44 @@ pub async fn verify_person_in_community(
   CommunityPersonBanView::check(&mut context.pool(), person_id, community_id).await
 }
 
+/// Fetches the person and community or site to verify their type, then checks if person is banned
+/// from local site or community.
+pub async fn verify_person_in_site_or_community(
+  person_id: &ObjectId<ApubPerson>,
+  site_or_community: &Either<ApubSite, ApubCommunity>,
+  context: &Data<LemmyContext>,
+) -> LemmyResult<()> {
+  let person = person_id.dereference(context).await?;
+  InstanceActions::check_ban(&mut context.pool(), person.id, person.instance_id).await?;
+  if let Either::Right(community) = site_or_community {
+    let person_id = person.id;
+    let community_id = community.id;
+    CommunityPersonBanView::check(&mut context.pool(), person_id, community_id).await?;
+  }
+  Ok(())
+}
+
+/// Verify that admin action was performed by an admin.
+///
+/// * `mod_id` - Activitypub ID of the admin who performed the action
+/// * `site` - The site that the person should be an admin of
+pub async fn verify_admin_action(
+  admin_id: &ObjectId<ApubPerson>,
+  site: &Site,
+  context: &Data<LemmyContext>,
+) -> LemmyResult<()> {
+  // admin action comes from the correct instance, so it was presumably done
+  // by an instance admin.
+  // TODO: federate instance admin status and check it here
+  if admin_id.inner().domain() == site.ap_id.domain() {
+    return Ok(());
+  }
+
+  let admin = admin_id.dereference(context).await?;
+  let local_user_view = LocalUserView::read_person(&mut context.pool(), admin.id).await?;
+  is_admin(&local_user_view)
+}
```

Comment:
This function is only called once inside of `verify_mod_or_admin_action` so you can move the logic directly there.